doc/rst/technotes/main.rst

```diff
@@ -36,89 +35,79 @@ For most uses, Chapel's existing config variable support is useful and
 other information explaining what non-config-var command line
 arguments are available and what they might do. For this reason, when
-a Chapel program includes a main declared to take arguments, ``--help``
-and ``-h`` will be passed on to the main(args) function.  The function
-Help.printUsage() will print out the familiar config var table.
+a Chapel program includes a ``main()`` declared to take arguments,
+``--help`` and ``-h`` will be passed on to the ``main(args)`` function.
 
-To gain access to this new function requires a module:
+The function ``printUsage()`` will print out the familiar config var
+table. To gain access to this new function requires a module:
 
-``use Help;``
+.. literalinclude:: ../../../test/technotes/doc-examples/MainProcExample0.chpl
+   :language: chapel
+   :start-after: START_EXAMPLE_1
+   :end-before: STOP_EXAMPLE_1
 
-From that module, the function printDefaultUsage() is available. For
-example, consider the following program:
-
-.. code-block:: chapel
-
-  use Help;
-
-  proc main(args: [] string) {
-    for a in args {
-      if a=="--help" {
-        printUsage();
-        writeln("\nEXTRA ARGUMENTS:");
-        writeln(  "================");
-        writeln("You can provide an arbitrary list of filenames.");
-        exit(0);
-      }
-    }
-  }
-
-This program scans its argument list for ``--help``, and if found, it
-prints the standard usage message as well as some additional text
+The following program scans its argument list for ``--help``, and if
+found, it prints the standard usage message as well as some additional text
 explaining the purpose of the extra arguments. Note that this could be
 used by a program which simply wants to add explanatory text to the
 ``--help`` output without actually handling any additional arguments.
 
-Finally, as is common in GNU programs, there is a way to suspend
-argument processing when main is declared to take arguments. Suppose
-you are writing ``rm`` in Chapel and need to delete a file named
-``--memTrack=true``. Your program will consume this flag during startup,
-rather than passing it on to the declared main function. However,
-placing the argument ``--`` in the argument list will cause all
-following arguments to be passed to the declared main function and
-will suspend handling of any builtin command line
-arguments. Specifically, if ``myprogram`` is a Chapel program which has
-declared main to take arguments:
+.. literalinclude:: ../../../test/technotes/doc-examples/MainProcExample1.chpl
+   :language: chapel
+   :start-after: START_EXAMPLE_0
+   :end-before: STOP_EXAMPLE_0
+
+There is a way to suspend argument processing when ``main()`` is declared
+to take arguments.
+
+Suppose that you need to pass a file named ``--memTrack=true`` to the
+program written above. If passed like normal, then the Chapel runtime
+will consume this argument at startup and use it to set the associated
+config var named ``memTrack``.
```

Comment:
Might be fun to link to `memTrack` here in case people are curious about it.  But I won't require it

Reply:
I think I'm gonna opt to skip linking it because it's in an unstable variable in an unstable internal module that's used by an unstable standard module, lol. Maybe this means I think of a better flag name or add my own `config const` instead.